pkg/querier/iterators/chunk_merge_iterator.go

```diff
@@ -34,7 +36,7 @@ func NewChunkMergeIterator(cs []chunk.Chunk, _, _ model.Time) chunkenc.Iterator
 	}
 
 	for _, iter := range c.its {
-		if iter.Next() {
+		if iter.Next() == chunkenc.ValFloat {
```

Comment:
Isn't safer to iterate until `!= chunkenc.ValNone` and then check the type at each iteration, populating `c.currErr` if the value is not a float, because not supported.

_Same comment applies below to `Seek()`._

Reply:
yes, also I think this iterator is not equivalent to the batched iterator in it's current form: https://github.com/grafana/mimir/blob/b777f85533a3f6084658eea38fc47dfdf501fb24/pkg/querier/querier.go#L93  :(
Since you also raised a performance concern with the batched iterator, I'm thinking that on this branch we should just error out there as well if an underlying chunk is of non-float

Reply:
this line should be fixed now